implementation/src/main/java/com/teampc/model/admin/User.java

```diff
@@ -143,5 +143,27 @@ public static User fromString(String serialized) {
       }
       return user;
    }
+
+   public UserType getUserType() {
+      return accept((Visitor<UserType>) new Visitor<UserType>() {
+         @Override
+         public UserType visitTeacher(Teacher t) {
+            return UserType.TEACHER;
+         }
+
+         @Override
+         public UserType visitStudent(Student s) {
+            return UserType.STUDENT;
+         }
+      });
+   }
+
+   public boolean isTeacher() {
```

Comment:
same deal here, ordinarily just make this abstract and have Teacher return true and Student return false